src/sagas/navigation.js

```diff
@@ -0,0 +1,27 @@
+import {takeLatest } from 'redux-saga/effects';
+
+import {ActionTypes} from '../common/constants';
+
+export function* goBack(action) {
+  try {
+    yield action.payload();
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+export function* goTo(action) {
+  try {
+    yield action.payload.method(action.payload.args);
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+export function* watchGoBack() {
```

Comment:
you can make single export here watchNavigationActions and have both `yield takeLatest(...)` lines in there.